Component.vue

```diff
@@ -0,0 +1,29 @@
+<script setup>
+import { ref } from "vue";
+import axios from "axios";
+const pageTitle = "Data Fetching Example";
+const isLoading = ref(false);
+const data = ref([]);
+const fetchData = async () => {
+  try {
+    isLoading.value = true;
+    const response = await axios.get("https://api.example.com/data");
+    data = response.data;
+  } catch (error) {
+    console.error("Error fetching data:", error);
+  } finally {
+    isLoading.value = false;
+  }
+};
+</script>
+
+<template>
+  <div>
+    <h1>{{ pageTitle }}</h1>
+    <button @click="fetchData">Fetch Data</button>
```

Comment:
i would think it's a good idea to disable the button while the data `isLoading`
```suggestion
    <button @click="fetchData" :disabled="isLoading">Fetch Data</button>
```